great_expectations/core/__init__.py

```diff
@@ -900,156 +1192,142 @@ def append_expectation(self, expectation_config):
         """
         self.expectations.append(expectation_config)
 
-    def find_expectation_indexes(
-        self, expectation_type=None, column=None, expectation_kwargs=None
-    ):
-        """Find matching expectations and return their indexes.
-        Args:
-            expectation_type=None                : The name of the expectation type to be matched.
-            column=None                          : The name of the column to be matched.
-            expectation_kwargs=None              : A dictionary of kwargs to match against.
-
-        Returns:
-            A list of indexes for matching expectation objects.
-            If there are no matches, the list will be empty.
+    def remove_expectation(
+        self,
+        expectation_configuration: ExpectationConfiguration,
+        match_type: str = "domain",
+    ) -> ExpectationConfiguration:
         """
-        if expectation_kwargs is None:
-            expectation_kwargs = {}
-
-        if (
-            "column" in expectation_kwargs
-            and column is not None
-            and column is not expectation_kwargs["column"]
-        ):
-            raise ValueError(
-                "Conflicting column names in find_expectation_indexes: %s and %s"
-                % (column, expectation_kwargs["column"])
-            )
 
-        if column is not None:
-            expectation_kwargs["column"] = column
+        Args:
+            expectation_configuration: This is the configuration for the expectation
```

Comment:
More verbose docs here: "A potentially incomplete (partial) Expectation Configuration to match against for removing expectaitons"